packages/apps/tools/src/components/Global/AddNetworkModal/AddNetworkModal.tsx

```diff
@@ -0,0 +1,114 @@
+import { Button, Stack, TextField, useModal } from '@kadena/react-ui';
+
+import {
+  formButtonStyle,
+  formContentStyle,
+  modalOptionsContentStyle,
+} from './styles.css';
+
+import { useWalletConnectClient } from '@/context/connect-wallet-context';
+import { zodResolver } from '@hookform/resolvers/zod';
+import useTranslation from 'next-translate/useTranslation';
+import type { FC } from 'react';
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import * as z from 'zod';
+
+// @see; https://www.geeksforgeeks.org/how-to-validate-a-domain-name-using-regular-expression/
+const DOMAIN_NAME_REGEX: RegExp =
+  /^(?!-)[A-Za-z0-9-]+([\-\.]{1}[a-z0-9]+)*\.[A-Za-z]{2,6}$/;
```

Comment:
The network host can be anything. If someone hosted it on a local machine, they could use a hostname, ip-address, or ipv6 as well, which all aren't FQDN. I don't think it should be limited in any way.